unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Avatar.cs

```diff
@@ -205,9 +216,9 @@ public virtual void AddVisibilityConstraint(string key)
         public void Dispose()
```

Comment:
One additional question. Is it safe to re-dispose elements that have been disposed?

Imagine this situation. The avatar gets disposed while loading, therefore the `OperationCanceledException` is catched. But inside there, items are re-disposed again; as they were in this `Dispose()` method. Could that be an issue?

Reply:
I agree with your comment, i think its an anti-pattern. I didnt want to change it because so far it has been working like this.

Reply:
In addition, i think that the proper way of improving this is by every object (loader, visibility, etc) handling its own `OperationCancelledException` avoiding an explicit disposal call from the `Avatar` class